emukit/quadrature/methods/warped_bq_model.py

```diff
@@ -40,88 +42,93 @@ def X(self) -> np.ndarray:
-        """
-        Computes predictive means and covariance of the warped GP as well as the base GP
+        """Compute predictive means and covariance of the warped GP as well as the base GP.
 
-        :param X_pred: Locations at which to predict, shape (n_points, input_dim)
-        :returns: predictive mean and covariance of warped GP, predictive mean and covariance of base-GP in that order.
-        mean shapes both (n_points, 1) and covariance shapes both (n_points, n_points)
+        :param X_pred: Locations at which to predict, shape (num_points, input_dim)
+        :returns: Predictive mean and covariance of warped GP, predictive mean and covariance of base-GP in that order.
+                  mean shapes both (num_points, 1) and covariance shapes both (num_points, num_points)
         """
         raise NotImplemented
 
     def predict_with_full_covariance(self, X_pred: np.ndarray) -> Tuple[np.ndarray, np.ndarray]:
-        """
-        Computes approximate predictive means and full co-variances of warped GP.
+        """Compute predictive means and covariance of warped GP.
 
         :param X_pred: Locations at which to predict, shape (n_points, input_dim)
         :return: predictive mean, predictive full covariance of warped-GP, shapes (n_points, 1) and (n_points, n_points)
         """
         return self.predict_base_with_full_covariance(X_pred)[:2]
 
     def predict(self, X_pred: np.ndarray) -> Tuple[np.ndarray, np.ndarray]:
-        """
-        Computes predictive means and variances of warped-GP.
+        """Compute predictive means and variances of warped-GP.
 
         :param X_pred: Locations at which to predict, shape (n_points, input_dim)
         :return: predictive mean, predictive variances of warped-GP, both shapes (n_points, 1)
         """
         return self.predict_base(X_pred)[:2]
 
     def set_data(self, X: np.ndarray, Y: np.ndarray) -> None:
+        """Set the new data in the model.
+
+        First, the model parameters that are not being optimized are updated, as they may depend on the new data,
+        then new data is set in the model.
+
+        :param X: Observation locations, shape (num_points, input_dim)
+        :param Y: Integrand observations at X, shape (num_points, 1)
         """
-        This method transforms the integrand y values and sets the data
-        :param X: observed locations
-        :param Y: observed integrand values
+        self.update_parameters(X, Y)
+        self.base_gp.set_data(X, self._warping.inverse_transform(Y))
+
+    def update_parameters(self, X: np.ndarray, Y: np.ndarray) -> None:
+        """Update parameters of the model that are not being optimized. Use pass if no parameters need to be updated.
+
+        :param X: Observation locations, shape (num_points, input_dim)
+        :param Y: Integrand observations at X, shape (num_points, 1)
         """
-        self.base_gp.set_data(X, self.inverse_transform(Y))
+        raise NotImplementedError
```

Comment:
Maybe you could just do `pass` here, and only those classes that need this override this method

Reply:
OK, done.